### PR TITLE
Allow Alluxio hadoop client uri to have empty authority

### PR DESCRIPTION
**Pull request by LuQQiu**

### CODE REVIEW

**Comment by madanadit** [comment]
@LuQQiu do you need someone to review this PR?

**Comment by LuQQiu** [comment]
@ns1123 PTAL, thanks!

**Comment by LuQQiu** [comment]
@ns1123 Addressed the comments and add the unit tests, PTAL

File: core/common/src/main/java/alluxio/exception/PreconditionMessage.java [hunk 1]
```diff
@@ -51,6 +51,7 @@
   ERR_ZK_ADDRESS_NOT_SET("Cannot get leader address from zookeeper; %s is not set"),
   FILE_TO_PERSIST_MUST_BE_COMPLETE("File being persisted must be complete"),
   FILE_WRITE_LOCATION_POLICY_UNSPECIFIED("The location policy is not specified"),
+  MASTER_HOSTNAME_NULL("Master hostname must not be null"),
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/exception/PreconditionMessage.java]: Removed the message here and give more detail error message.

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 2]
```diff
@@ -505,10 +505,15 @@ void initializeInternal(URI uri, org.apache.hadoop.conf.Configuration conf) thro
     // When using zookeeper we get the leader master address from the alluxio.zookeeper.address
     // configuration property, so the user doesn't need to specify the authority.
     if (!Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
-      Preconditions.checkNotNull(uri.getHost(), PreconditionMessage.URI_HOST_NULL);
-      Preconditions.checkState(uri.getPort() != -1, PreconditionMessage.URI_PORT_NULL);
-      Configuration.set(PropertyKey.MASTER_HOSTNAME, uri.getHost());
-      Configuration.set(PropertyKey.MASTER_RPC_PORT, uri.getPort());
+      if (uri.getAuthority() != null) {
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: Add two unit tests, thanks

**Comment by ns1123** [comment]
@LuQQiu  LGTM

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**